src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -50,25 +43,25 @@ object TezosPlatformDiscoveryOperations {
 /** Class providing the implementation of the metadata calls with caching */
 class TezosPlatformDiscoveryOperations(
   metadataOperations: MetadataOperations,
-  attributesCache: MVar[IO, AttributesCache],
-  entitiesCache: MVar[IO, EntitiesCache],
-  attributeValuesCache: MVar[IO, AttributeValuesCache],
+  caching: MetadataCaching[IO],
```

Comment:
no, I'm against it, because it would mean that we have global mutable state available directly without any idea how it got there. Look at the DbUtils, you can randomly import db, and run query in any class/object in the code. I disagree with this approach.

Reply:
Maybe I wasn't too clear.
I simply meant that, since nobody outside of this class seems to use the instance passed in as a caching parameter, we might completely encapsulate the fact that a cache is built, create it with the PlatformDiscoveryOps, and use it only internally

Reply:
At this point I think I like cache being used explicitly. Maybe when I'll try to move more logic to cache it will be more appropriate, but we'll see.